plugins/storage/volume/adaptive/src/main/java/org/apache/cloudstack/storage/datastore/lifecycle/AdaptiveDataStoreLifeCycleImpl.java

```diff
@@ -404,4 +404,14 @@ public void disableStoragePool(DataStore store) {
         s_logger.info("Disabling storage pool [" + store.getName() + "]");
         _dataStoreHelper.disable(store);
     }
+
+    @Override
+    public boolean changeStoragePoolScopeToZone(DataStore store, ClusterScope clusterScope, HypervisorType hypervisorType) {
```

Comment:
the same code has been used in many classes.
is it better to inherit from a new class like `AbstractDataStoreLifeCycleImpl.java` ?

Reply:
Thanks for the idea. I have implemented it this way.